File: Contests/Chef_MARCH20B/ENGXOR.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
ll oddSetBits(ll n){
    if (n == 0) return 0;
    if (n%2 == 0) return (oddSetBits(floor(n/2)));
    if (n%2 == 1) return (1 - oddSetBits(floor(n/2)));
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    ll tc;
    cin >> tc;
    for(ll test=0; test<tc; test++){
        ll n,q;
        cin >> n >> q;
        ll arr[n];
        for(ll i=0; i<n; i++){
            cin >> arr[i];
        }
        for(ll qr=0; qr<q; qr++){
            ll elem;
            cin >> elem;
            ll odd = 0;
            ll even = 0;
            for(ll i=0; i<n; i++){
                if (oddSetBits(elem ^ arr[i])==1) odd++;
                else even++;
            }
            cout << even << " " << odd << "\n";
        }
    }
}

File: Contests/Chef_MARCH20B/ENGXOR.py
def oddSetBits(n):
    # returns 1 if n has odd number of set bits, else 0 if n has even number of set bits
    if n == 0:
        return 0
    if n%2 == 0:
        return oddSetBits(n//2)
    if n%2 == 1:
        return 1 - oddSetBits(n//2)

def checkOddSetBits(n):
    # this is O(set bits) time
    cnt = 0
    while n>0:
        n = n & (n-1)
        cnt += 1
    return cnt%2

def main():
    for _ in range(int(input())):
        n,q = [int(x) for x in input().strip().split()]
        arr = [int(x) for x in input().strip().split()]
        for query in range(q):
            elem = int(input().strip())
            odd, even = 0, 0
            for i in arr:
                if checkOddSetBits(elem ^ i):
                    odd += 1
                else:
                    even += 1
            print(even,odd)
    
main()
        